File: Admin_p/Admin_p.Web/Modules/ServerTypes/electroDB/OrderDetailsColumns.ts
import { ColumnsBase, fieldsProxy } from "@serenity-is/corelib";
import { Column } from "@serenity-is/sleekgrid";
import { OrderDetailsRow } from "./OrderDetailsRow";

export interface OrderDetailsColumns {
    OrderDetailId: Column<OrderDetailsRow>;
    OrderStatus: Column<OrderDetailsRow>;
    ProductName: Column<OrderDetailsRow>;
    Quantity: Column<OrderDetailsRow>;
    UnitPrice: Column<OrderDetailsRow>;
    Subtotal: Column<OrderDetailsRow>;
    CreatedAt: Column<OrderDetailsRow>;
    UpdatedAt: Column<OrderDetailsRow>;
}

export class OrderDetailsColumns extends ColumnsBase<OrderDetailsRow> {
    static readonly columnsKey = 'electroDB.OrderDetails';
    static readonly Fields = fieldsProxy<OrderDetailsColumns>();
}

File: Admin_p/Admin_p.Web/Modules/electroDB/OrderDetails/OrderDetailsColumns.cs
using Serenity.ComponentModel;
using System;
using System.ComponentModel;

namespace Admin_p.electroDB.Columns;

[ColumnsScript("electroDB.OrderDetails")]
[BasedOnRow(typeof(OrderDetailsRow), CheckNames = true)]
public class OrderDetailsColumns
{
    [EditLink, DisplayName("Db.Shared.RecordId"), AlignRight]
    public int OrderDetailId { get; set; }
    public string OrderStatus { get; set; }
    public string ProductName { get; set; }
    public int Quantity { get; set; }
    public decimal UnitPrice { get; set; }
    public decimal Subtotal { get; set; }
    public DateTime CreatedAt { get; set; }
    public DateTime UpdatedAt { get; set; }
}